l1-contracts/test/validator-selection/ValidatorSelection.t.sol

```diff
@@ -233,79 +255,172 @@ contract ValidatorSelectionTest is ValidatorSelectionTestBase {
-        proposerNotProvided: false,
-        invalidCommitteeCommitment: true
+        senderIsNotProposer: true,
+        proposerAttestationNotProvided: false,
+        invalidAttestation: false,
+        invalidSigners: true
       })
     );
   }
 
-  function testInsufficientSigsMove() public setup(4, 4) progressEpochs(2) {
+  function testInvalidAttestation() public setup(4, 4) progressEpochs(2) {
+    ProposeTestData memory ree = _testBlock("mixed_block_1", NO_REVERT, 3, 4, INVALID_ATTESTATION);
+
+    // the invalid attestation is the first one
+    _invalidateByAttestationSig(ree, 1, Errors.Rollup__AttestationsAreValid.selector);
+    _invalidateByAttestationSig(ree, 0, NO_REVERT);
+  }
+
+  function testInsufficientSignatures() public setup(4, 4) progressEpochs(2) {
+    ProposeTestData memory ree = _testBlock("mixed_block_1", NO_REVERT, 2, 4, NO_FLAGS);
+
+    _invalidateByAttestationCount(ree, NO_REVERT);
+  }
+
+  function testInvalidateMultipleBlocks() public setup(4, 4) progressEpochs(2) {
+    uint256 initialBlockNumber = rollup.getPendingBlockNumber();
+    ProposeTestData memory ree = _testBlock("mixed_block_1", NO_REVERT, 3, 4, INVALID_ATTESTATION);
+    _testBlock("mixed_block_2", NO_REVERT, 3, 4, NO_FLAGS);
+
+    _invalidateByAttestationSig(ree, 0, NO_REVERT, initialBlockNumber + 1);
+  }
+
+  function testProposeBlockAfterInvalidate() public setup(4, 4) progressEpochs(2) {
+    uint256 initialBlockNumber = rollup.getPendingBlockNumber();
+    ProposeTestData memory ree = _testBlock("mixed_block_1", NO_REVERT, 3, 4, INVALID_ATTESTATION);
+    _invalidateByAttestationSig(ree, 0, NO_REVERT);
+
+    _testBlock("mixed_block_1", NO_REVERT, 3, 4, NO_FLAGS);
+    assertEq(
+      rollup.getPendingBlockNumber(),
+      initialBlockNumber + 1,
+      "Failed to propose block after invalidate"
+    );
+  }
+
+  function testCannotProposeIfAllValidatorsHaveMoved() public setup(4, 4) progressEpochs(2) {
+    // Buried in the RollupBuilder, we add initial validators using l1-contracts/src/mock/MultiAdder.sol
+    // In that, you see this inconspicuous true at the end of the call to deposit. This means that the
+    // validators are going into the "bonus" instance, are are thus not tied directly to the rollup from
+    // the perspective of the GSE.
+    // The "bonus" instance (which is tracked by the GSE) is only available to the latest rollup in the GSE.
+    // So when we add a new 0xdead rollup, all the validators we added get moved over to that one,
+    // and our original rollup has no validators.
+    // So this is showing that in that case, even if all your validators move over, you still cannot build
+    // a block if you submit one with no signatures. This was a change from prior behavior where we had had
+    // that if there were zero validators in a rollup, anyone could build a block
+
+    // TODO(palla): What should we do in this scenario? Block the proposal, or allow invalidating later?
```

Comment:
The committee is still the same (why it was revert=true before), so as long as that committee is the active committee you kinda gotta wait. As it could also just be a partial move, you are in this weird state where the chain is degraded. 
I believe you will also have a very similar thing going on at the new rollup, because of the delayed reads (2 epochs) the first two epochs after attesters were added there won't be much to do 👀